Abstract Syntax and Semantics Document/index.html

```diff
@@ -500,53 +500,159 @@
-    
-    <h3>Count-based window functions</h3>
-    
-    <p><code>x = (l)</code>, where <code>l ∈ N</code>. The function <code>w<sub>#</sub>(S,p,t,x)</code> selects a substream <code>S<sub>1</sub></code> of <code>S</code> based on the time instant <code>t'</code> satisfying that:</p>
-    
-    <ul>
-      <li>for every <code>t' &lt; t'' ≤ t</code>, there are fewer than <code>l</code> timestamped graphs associated with predicate <code>p</code> in <code>S</code> from <code>t''</code> to <code>t</code>,</li>
-      <li>there are at least <code>l</code> timestamped graphs associated with predicate <code>p</code> in <code>S</code> from <code>t'</code> to <code>t</code>.</li>
-    </ul>
-    
-    <p>Elements from <code>S<sub>1</sub></code> are those <code>((n<sub>i</sub>,g<sub>i</sub>),p,t<sub>i</sub>)</code> from <code>S</code> having <code>t<sub>i</sub> ≥ t'</code>. In case there are more than <code>l</code> timestamped graphs associated with predicate <code>p</code> in <code>S</code> from <code>t'</code> to <code>t</code>, only timestamped graphs at <code>t'</code> are removed at random.</p>
-    
-    <p>Formally:</p>
-    
-    <p><code>w<sub>#</sub>(S,p,t,(l)) = {((n,g),p,t'') ∈ S ∣ t' &lt; t'' ≤ t} ∪ X(S|<sub>p</sub>[t',t'], l - #S|<sub>p</sub>[t'+1,t])</code>,</p>
-    
-    <p>where</p>
-    
-    <ul>
-      <li><code>S|<sub>p</sub>[t<sub>1</sub>,t<sub>2</sub>] = {((n,g),p,t'') ∈ S ∣ t<sub>1</sub> ≤ t'' ≤ t<sub>2</sub>}</code>,</li>
-      <li><code>#S|<sub>p</sub>[t<sub>1</sub>,t<sub>2</sub>]</code> is the number of elements of this set,</li>
-      <li><code>t'</code> satisfies that <code>#S|<sub>p</sub>[t',t] ≥ l</code> and <code>#S|<sub>p</sub>[t'+1,t] &lt; l</code> </li>
-    </ul>
+      <p class="note">The substream resulting from the application of a time-based window function is time-bounded.</p>
     </section>
-    
+      <section id="temporal-count-based-window-functions">
+      <h3>Temporal-count-based window functions</h3>
+      <p>A <em>temporal-count-based window function</em> <code>w</code> is a window function specified by
+        a finite set <code>w<sub>P</sub></code> of timestamp predicates together with
+        semi-infinite temporal intervals <code>w<sub>J</sub>(P)</code> of each timestamp predicate <code>P</code> in <code>w<sub>P</sub></code>
+        with endpoints <code>w<sub>T</sub>(P)</code>,
+        and positive integers <code>w<sub>N</sub>(P)</code>, for each timestamp predicate <code>P</code> in <code>w<sub>P</sub></code> such that
+        for every stream <code>S</code>, 
+        an element <code>S(i)</code> of <code>S</code> is a member of <code>w(S)</code> if and only if
+        the timestamp predicate  <code>P</code> of <code>S(i)</code> is in <code>P</code> and
+        <ul>          
+          <li><code>t</code> is the timestamp of <code>S(i)</code>,</li>
+          <li><code>T</code> is the set of temporal entities that are timestamps for elements of <code>S</code>
```

Comment:
Since the set of timestamps in a stream S is a dependent function of the stream S, we could us T(S) instead of T here. Do you think that would be clear enough?

We could also introduce a more special symbol for the set (should actually be class, I think) of all temporal entities, instead of "T". E.g. T<sub>all</sub>